drivers/fuel_gauge/fuel_gauge_syscall_handlers.c

```diff
@@ -46,3 +47,27 @@ static inline int z_vrfy_fuel_gauge_set_prop(const struct device *dev,
 }
 
 #include <syscalls/fuel_gauge_set_prop_mrsh.c>
+
+static inline int z_vrfy_fuel_gauge_get_buffer_prop(const struct device *dev,
+						    struct fuel_gauge_get_buffer_property *prop,
+						    void *dst, size_t dst_len)
+{
+	char k_buf[dst_len];
+	struct fuel_gauge_get_buffer_property k_prop;
+
```

Comment:
I don't think we need to copy into a stack param then into the output dst param

Instead the following should be enough for the output buffer. The property still needs to be copied as it likely has branch logic on it!

Z_OOPS(Z_SYSCALL_MEMORY_WRITE(dst, dst_len));

z_impl_fuel_gauge_get_buffer_prop(dev, &k_prop, dst, dst_len);

Reply:
Fixed and tested - thanks for the pointer